libraries/AP_HAL_PX4/RCOutput.cpp

```diff
@@ -598,17 +618,26 @@ void PX4RCOutput::set_output_mode(enum output_mode mode)
         set_freq(_rate_mask, 1);
     }
     _output_mode = mode;
-    if (_output_mode == MODE_PWM_ONESHOT) {
-        //::printf("enable oneshot\n");
+    switch (_output_mode) {
+    case MODE_PWM_ONESHOT:
         ioctl(_pwm_fd, PWM_SERVO_SET_ONESHOT, 1);
         if (_alt_fd != -1) {
             ioctl(_alt_fd, PWM_SERVO_SET_ONESHOT, 1);
         }
-    } else {
+        break;
+    case MODE_PWM_NORMAL:
         ioctl(_pwm_fd, PWM_SERVO_SET_ONESHOT, 0);
         if (_alt_fd != -1) {
             ioctl(_alt_fd, PWM_SERVO_SET_ONESHOT, 0);
         }
+        break;
+    case MODE_PWM_BRUSHED16KHZ:
+        // setup an 8MHz clock. This has the effect of scaling all outputs by 8x
+        ioctl(_pwm_fd, PWM_SERVO_SET_UPDATE_CLOCK, 8);
+        if (_alt_fd != -1) {
+            ioctl(_alt_fd, PWM_SERVO_SET_UPDATE_CLOCK, 8);
+        }
```

Comment:
So I was going to write if we shouldn't change the clock back to 1 as the MODE_PWM_NORMAL changes OneShot back to 0, but then I realized that this function is called only at initialization. I think the MOT_PWM_TYPE parameter needs to be marked as restart required.